test/k8s/datapath_configuration.go

```diff
@@ -1000,6 +1000,9 @@ func testPodHTTPToOutside(kubectl *helpers.Kubectl, outsideURL string, expectNod
 	deploymentManager.WaitUntilReady()
 
 	label := "zgroup=testDSClient"
+	err := kubectl.WaitforPods(namespace, "-l "+label, helpers.HelperTimeout)
+	ExpectWithOffset(1, err).Should(BeNil(), "Failed to wait for pods with labels: %s", label)
```

Comment:
We should leave a comment saying why we are checking pod ready status again given that `WaitUntilReady` calls the same [underlying function](https://github.com/cilium/cilium/blob/636865942b3b2d7e4bd666a16db1ba57f78c8b16/test/helpers/kubectl.go#L1434) that `WaitforPods` does, so it may be confusing to readers.